### PR TITLE
UCR jsonpath expression

### PR DESCRIPTION
**Pull request by snopoke**
## Product Description
Adds a `jsonpath` UCR expression.

See docs [inline](https://github.com/dimagi/commcare-hq/pull/31646/files#diff-bb6b73c3251b764dd76b43fbe99e34a32ae3bb86ca78799ca1d1cc611b815eadR462).

No real use case for this right now but I think it's potentially useful for working with complex data structures.

## Technical Summary
UCR expression that evaluates a jsonpath expression on the doc and returns the result.

1st commit is just sorting lines.

## Feature Flag
Anything that exposes raw UCR expressions:
* USER_CONFIGURABLE_REPORTS
* EXPRESSION_REPEATER
* UCR_EXPRESSION_REGISTRY

(it's a growing list!)

### Safety story
Few feature not currently in use. Good test coverage.

### Automated test coverage
Added tests for this new expression

### QA Plan
None

### Migrations
NA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by proteusvacuum — APPROVED** [review]
Super cool! This is a much a clearer way of expressing array index and property path expressions, and should probably be used in their place.

File: corehq/apps/userreports/expressions/specs.py [hunk 1]
```diff
@@ -454,6 +457,79 @@ def __str__(self):
...
+class JsonpathExpressionSpec(NoPropertyTypeCoercionMixIn, JsonObject):
+    """
+    This will execute the jsonpath expression against the current doc
+    and emit the result.
+
+    .. code:: json
+
+       {
+           "type": "jsonpath",
+           "jsonpath": "form..case.name",
```
[the remark this answers is not in the document]
Comment (snopoke) [reply corehq/apps/userreports/expressions/specs.py:469]: The `..` is intentional. It's a recursive descent operator.

**Comment by snopoke** [comment]
> Super cool! This is a much a clearer way of expressing array index and property path expressions, and should probably be used in their place.

I mostly agree. The only challenge is that it always returns a list which is a but frustrating.

**Comment by snopoke** [comment]
> > Super cool! This is a much a clearer way of expressing array index and property path expressions, and should probably be used in their place.
> 
> I mostly agree. The only challenge is that it always returns a list which is a bit frustrating. I toyed with ways to coerce it to return single values but it would require configuration from the user to know when that's appropriate or not.

Fixed this issue here: 9e20389dff8

I realised you can specify the `datatype` as `array` if you specifically need a list as the result.

**Pull request merged**